src/pylint_odoo/checkers/import_checker.py

```diff
@@ -0,0 +1,128 @@
+import re
+from typing import Union
+
+from astroid import Import, ImportFrom, Module
+from astroid.modutils import modpath_from_file
+from pylint.checkers.utils import only_required_for_messages
+from pylint.lint import PyLinter
+
+from .odoo_base_checker import OdooBaseChecker
+
+ODOO_MSGS = {
+    # C->convention R->refactor W->warning E->error F->fatal
+    "E8401": (
+        "This python module is not imported and has no effect",
+        "unused-module",
```

Comment:
IMHO this lint name is not specific enough. What kind of module? Take into account there are Odoo modules and Python modules

I'd say Odoo models instead, e.g. `odoo-model-not-imported`.

What do you think?

Reply:
Initially it was for odoo models only but according to the previous review by Moy I made it generic. By Python module I think I am following the official specification which as far as I understand is any `.py` file.

As of now this basically detects unused python files